tensorflow/compiler/jit/xla_gpu_device.cc

```diff
@@ -52,8 +53,34 @@ Status XlaGpuDeviceFactory::CreateDevices(
     VLOG(1) << "Failed to create XLA_GPU device: " << platform.status();
     return Status::OK();
   }
-
-  for (int i = 0; i < platform.ValueOrDie()->VisibleDeviceCount(); ++i) {
+  const auto& allowed_gpus =
+      session_options.config.gpu_options().visible_device_list();
+  std::unordered_set<int> gpu_ids;
+  int num_visible_devices = platform.ValueOrDie()->VisibleDeviceCount();
+  if (allowed_gpus.empty()) {
+    for (int i = 0; i < num_visible_devices; ++i) gpu_ids.insert(i);
+  } else {
+    const std::vector<string> visible_devices =
+        str_util::Split(allowed_gpus, ',');
+    // copied from gpu/gpu_device.cc Should be redundant since code would fail
+    // there before it gets to here.
+    for (const string& platform_gpu_id_str : visible_devices) {
+      int32 platform_gpu_id;
+      if (!strings::safe_strto32(platform_gpu_id_str, &platform_gpu_id)) {
```

Comment:
Please use absl rather than TF strings code, we avoid that now within XLA.

Reply:
done